tests/packaging-suite/packaging.tests.js

```diff
@@ -0,0 +1,124 @@
+    const templateName = 'aws-nodejs';
+    execSync(`${serverlessExec} create --template ${templateName}`, { cwd });
+    execSync(`${serverlessExec} package`, { cwd });
+    return testUtils.listZipFiles(path.join(cwd, '.serverless/aws-nodejs.zip'))
+      .then(zipfiles => {
+        expect(zipfiles).toEqual(['handler.js']);
+      });
+  });
+
+  it('packages the default aws template with an npm dep correctly in the zip', () => {
+    const templateName = 'aws-nodejs';
+    execSync(`${serverlessExec} create --template ${templateName}`, { cwd });
+    execSync('npm init --yes', { cwd });
+    execSync('npm i lodash', { cwd });
+    execSync(`${serverlessExec} package`, { cwd });
+    return testUtils.listZipFiles(path.join(cwd, '.serverless/aws-nodejs.zip'))
+      .then(zipfiles => {
+        const nodeModules = new Set(
+          zipfiles.filter(f => f.startsWith('node_modules')).map(f => f.split(path.sep)[1]));
+        const nonNodeModulesFiles = zipfiles.filter(f => !f.startsWith('node_modules'));
+        expect(nodeModules).toEqual(new Set(['lodash']));
+        expect(nonNodeModulesFiles).toEqual(['handler.js', 'package-lock.json', 'package.json']);
+      });
+  });
+
+  it('doesn\'t package a dev dependency in the zip', () => {
+    const templateName = 'aws-nodejs';
+    execSync(`${serverlessExec} create --template ${templateName}`, { cwd });
+    execSync('npm init --yes', { cwd });
+    execSync('npm i --save-dev lodash', { cwd });
+    execSync(`${serverlessExec} package`, { cwd });
+    return testUtils.listZipFiles(path.join(cwd, '.serverless/aws-nodejs.zip'))
+      .then(zipfiles => {
+        const nodeModules = new Set(
+          zipfiles.filter(f => f.startsWith('node_modules')).map(f => f.split(path.sep)[1]));
+        const nonNodeModulesFiles = zipfiles.filter(f => !f.startsWith('node_modules'));
+        expect(nodeModules).toEqual(new Set([]));
+        expect(nonNodeModulesFiles).toEqual(['handler.js', 'package-lock.json', 'package.json']);
+      });
+  });
+
+  it('ignores package json files per ignore directive in the zip', () => {
+    const templateName = 'aws-nodejs';
+    execSync(`${serverlessExec} create --template ${templateName}`, { cwd });
+    execSync('npm init --yes', { cwd });
+    execSync('echo \'package: {exclude: ["package*.json"]}\' >> serverless.yml', { cwd });
+    execSync('npm i lodash', { cwd });
+    execSync(`${serverlessExec} package`, { cwd });
+    return testUtils.listZipFiles(path.join(cwd, '.serverless/aws-nodejs.zip'))
+      .then(zipfiles => {
+        const nodeModules = new Set(
+          zipfiles.filter(f => f.startsWith('node_modules')).map(f => f.split(path.sep)[1]));
+        const nonNodeModulesFiles = zipfiles.filter(f => !f.startsWith('node_modules'));
+        expect(nodeModules).toEqual(new Set(['lodash']));
+        expect(nonNodeModulesFiles).toEqual(['handler.js']);
+      });
+  });
+});
+
+describe('Integration test - Packaging', () => {
```

Comment:
I think we can nest this and rename this to `CloudFormation files` or something like that. So the structure could look like this:

```
describe('Integration Test - Packaging', () => {
  // the regular / core tests
  describe('CloudFormation files', () => {
    // CloudFormation specific tests
  })
})
```

This way we have an easy to read and reason about log output.

Reply:
I actually removed the separation since i have tests that check both.